lib/widgets/search-friends.dart

```diff
@@ -1,15 +1,150 @@
+                    border: OutlineInputBorder(), hintText: "Search by email")),
+          ),
+          noFound ? NoFoundMessage() : SizedBox.shrink(),
+          result != null ? FoundResult(result: result) : SizedBox.shrink(),
+        ],
+      ),
+    );
+  }
+
+  searchFriend(value) async {
+    DataSnapshot snapshot = await databaseReference
+        .child("index")
+        .orderByChild('email')
+        .equalTo(value)
+        .once();
+
+    if (snapshot.value != null) {
+      var uid = snapshot.value.keys.first;
+
+      setState(() {
+        result = Map.from({
+          "name": snapshot.value[uid]['name'],
+          "email": snapshot.value[uid]['email'],
+          'uid': uid
+        });
+        noFound = false;
+      });
+    } else {
+      setState(() {
+        noFound = true;
+        result = null;
+      });
+    }
+  }
+}
+
+class NoFoundMessage extends StatelessWidget {
+  @override
+  Widget build(BuildContext ctx) {
+    return Center(
+      child: Column(
+        children: <Widget>[
+          Text(
+            "Sorry...",
+            style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
+          ),
+          Text("There is no matching email.")
+        ],
+      ),
+    );
+  }
+}
+
+class FoundResult extends StatefulWidget {
+  FoundResult({@required this.result});
 
-    return Text("Add friend");
+  final Map result;
+
+  @override
+  _FoundResultState createState() => _FoundResultState();
+}
+
+class _FoundResultState extends State<FoundResult> {
+  final databaseReference = FirebaseDatabase.instance.reference();
+  final FirebaseAuth firebaseAuth = FirebaseAuth.instance;
+
+  bool confirmation = false;
+
+  @override
+  Widget build(BuildContext ctx) {
+    return confirmation
+        ? Padding(
+            padding: EdgeInsets.only(top: 16),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                Icon(Icons.check, size: 24),
+                Text('Request Sent!')
+              ],
+            ))
+        : Column(
+            children: <Widget>[
+              Text("Search Results...", style: TextStyle(fontSize: 16)),
+              ListTile(
+                title: Text(widget.result['name']),
+                subtitle: Text(widget.result['email']),
+                trailing: FlatButton(
+                  child:
+                      Text('Add Friend', style: TextStyle(color: Colors.blue)),
+                  onPressed: addFriend,
+                ),
+              )
+            ],
+          );
+  }
+
+  addFriend() async {
+    FirebaseUser currentUser = await firebaseAuth.currentUser();
+
+    DataSnapshot currentUserSnapshot = await databaseReference
+        .child("index")
+        .orderByChild('email')
+        .equalTo(currentUser.email)
+        .once();
+
+    await databaseReference
+        .child("users")
+        .child(widget.result['uid'])
+        .child("requests")
```

Comment:
This gets blocked by the database rules. Needs additional .child(currentUser.uid).set({name: name, email: email})